docs/proposals/pod-security-context.md

```diff
@@ -0,0 +1,302 @@
+        value of the deprecated field
+    3.  If the deprecated field and new field are both set, their values must match
+3.  For fields that are currently part of the `SecurityContext` **and are not** in the
+    `ContainerSecurityContext`:
+    1.  If the new field is set and the deprecated field is unset, the deprecated field is defaulted
+        to the value of the new field in all containers in the pod
+    2.  If the deprecated field is set and has the **same value** in each container and the new
+        field is unset, the new field is set to the value of the deprecated fields
+    3.  If the deprecated field is set and has more than a single value across all containers in a
+        pod, the new field remains unset and the deprecated field takes precedent
+4.  For fields that are currently in the `SecurityContext` **and are** in the
+    `ContainerSecurityContext`:
+    1.  If the field **is set** in the `ContainerDefaults` field of the `PodSecurityContext`, for
+        each container the following conditions apply:
+        1.  If the field is not set in the container's `SecurityContext` or
+            `ContainerSecurityContext` of the container, the deprecated field of the
+            `SecurityContext` is set to the value of the field in the container defaults of the
+            `PodSecurityContext`
+        2.  If the field is set in the `SecurityContext` and **not set** in the container's
+            `ContainerSecurityContext`, the field of the `ContainerSecurityContext` is set to the
+            value of the field in the container defaults of the `PodSecurityContext`
+        3.  If the field is set in the `ContainerSecurityContext` and **not set** in the
+            `SecurityContext`, the field of the `SecurityContext` is set to the value of the field
+            in the container defaults of the `PodSecurityContext`
+        4.  If the field is set in both the `SecurityContext` and `ContainerSecurityContext`, the
+            values of the fields must match
+    2.  If the field **is not set** in the `ContainerDefaults` field of the `PodSecurityContext`,
+        for each container the following conditions apply:
+        1.  If the field is not set in the container's `SecurityContext` or
+            `ContainerSecurityContext` of the container, no field values are altered
+        2.  If the field is set in the `SecurityContext` and **not set** in the container's
+            `ContainerSecurityContext`, the field of the `ContainerSecurityContext` is set to the
+            value of the field in the `SecurityContext`
+        3.  If the field is set in the `ContainerSecurityContext` and **not set** in the
+            `SecurityContext`, the field of the `SecurityContext` is set to the value of the field
+            in the `ContainerSecurityContext`
+        4.  If the field is set in both the `SecurityContext` and `ContainerSecurityContext`, the
```

Comment:
@thockin @smarterclayton 

> What if you made the PodSecurityContext a "macro" that gets overlayed with each container's security context and then cleared at POST time.

This was essentially the behavior I had originally wanted, compatibility issues aside.  I'm convinced now that we don't need the new `ContainerSecurityContext` type, which simplifies things a lot.